pages/index.tsx

```diff
@@ -18,7 +18,7 @@ const Home = () => {
           .get()
           .then(snapshot =>
             snapshot.forEach(doc => {
-              setItems(prev => [...prev, { ...doc.data(), key: doc.id } as IItem]);
+              setItems(prev => [...prev, doc.data() as IItem]);
```

Comment:
here we probably need `as IItem` because ts doesn't know `doc.data()` is item

Reply:
Basically when we use `as [sometype]` (this is called casting). we should really consider if it's really needed. 
Best case we don't want to do any casting. In this case ideally we should do deserialisation or like